### ISSUE TITLE
Accessibility Improvements Around Alt-Text/Captions - Add on upload and Multiline

### ISSUE THREAD

**Issue by shiribailem** [friendica/friendica#13062]
### Is the feature request related to a problem? Please describe.

Two paired elements around accessibility. When uploading images to posts there's no option to add the caption/alt-text before posting outside of manually editing the BB-Code except when you go out of your way to upload first to the photo library and then explicitly reshare from there.

Additionally, because those captions are often for meme/screenshot posts of text. It's a problem that the caption field is single-line instead of multi-line text entry. It would be beneficial to make it multi-line so that captions/alt-text can be made more readable (Also important to remember that alt-text isn't *just* for screenreaders, it can also be for things like allowing people to read it in more accessible fonts)

### Describe the feature you'd like

A caption edit button or field attached to the image upload box.

The caption field to be changed to multi-line input.

### Describe alternatives you've considered

As mentioned before the extra steps of doing a separate upload to specifically create the caption, or manually editing the bb-code to add the caption at the time of posting.

### Additional context

**Comment by MrPetovan** [comment]
Second part is addressed in https://github.com/friendica/friendica/issues/13005 which should itself be addressed in #12710 

For the first part, we recently added a quick way to attach images to posts via drag&drop or copy&paste, I don't suppose you are referring to these, but the file browser, right?

**Comment by shiribailem** [comment]
Definitely the drag/drop and copy/paste, it inserts the image and the bb-code, but there's nothing there to insert a caption at the same time.

The only way I've found is going into the photos section and editing the photo after upload.

**Comment by shiribailem** [comment]
Oh, and the browser tab for uploading also doesn't have an option either. I click an image and it just attaches it with the BB-Code, but nothing to add a caption outside of editing the BB-Code.

**Comment by MrPetovan** [comment]
Thank you for the elaboration, I don't think we'll be able to add image description input to the drag&drop and copy&paste feature, but we could probably add it to the file browser.

**Comment by annando** [comment]
What we can do is to improve the BBCode that is generated by the "drag&drop" feature. By now you have to not only add the alt description, but before this you have to change the format of the `img` element. This is of course only a small help.

Concerning multi line: Have you tried entering a multi line alt description?

**Comment by shiribailem** [comment]
@annando From the UI, yes, it didn't take any of the newlines. I know multiline *works*, but the element isn't multiline so it strips it. (I wouldn't be surprised if it was as simple as editing the html element)

**Comment by MrPetovan** [comment]
> Concerning multi line: Have you tried entering a multi line alt description?

I expect it to give this result: https://github.com/friendica/friendica/issues/13005

**Comment by anubis2814** [comment]
This is what it reads now.
 [url=https://friendica.myportal.social/photos/user/image/image][img]https://friendica.myportal.social/photo/image.png[/img][/url] 

A quick fix would be to change it to this
 [url=https://friendica.myportal.social/photos/user/image/image][img=https://friendica.myportal.social/photo/image.png]Add alt-text here[/img][/url] 

I could do it but I want to make sure this isn't already being done.

**Comment by MrPetovan** [comment]
Weird, it should be the following:
```
[url=https://friendica.myportal.social/photos/user/image/image]
[img=https://friendica.myportal.social/photo/image.png][/img]
[/url]
```
Notice how the image URL is now an attribute of the `img` BBCode.

**Comment by anubis2814** [comment]
It is in the dev code its not in the stable version.  So its been fixed just going to have to wait for the update.  Including a "enter alt-text here" I think would be good and make it easier for people who don't want to learn the bbcode for it..

**Comment by MrPetovan** [comment]
My contention with this is that a lot of pictures will have “Enter the alt-text here” as the only alt-text which isn’t much better.

**Comment by anubis2814** [comment]
Yeah it could come off as too judgy and self-righteous, trying to shame them

**Comment by shiribailem** [comment]
that's addressable by maybe a simple routine checking for "]Add alt-text here[" and replacing it with "][" on post?

**Comment by anubis2814** [comment]
That would make sense can't be that hard.

**Comment by MrPetovan** [comment]
It is rendered slightly more complicated by the fact that this string would need to be translated, which means the BBCode conversion method would need to be dependent on `L10n` which it had avoided so far.

My best idea to circumvent this issue that would make testing the conversion harder would be to pass the translated “Add alt-text here” placeholder string as an additional parameter to the `BBCode::convert` method, and figure out the dependency in the calling code.

**Comment by anubis2814** [comment]
An alternative that may or may not be easier could be popup asking if you want alt-text and if yes it inserts it in there, or shows them where in the code to insert it in the popup.  You could also have a checkbox that says "don't ask again" once they get it.  Its super easy once you know

**Comment by MrPetovan** [comment]
I thought about it, but the plugin we use for Drag-and-Drop allows for multiple images to be inserted at once, which would make these pop-ups absolutely unbearable if not utterly confusing. How would you know which image you’re describing when you uploaded three at a time?

**Comment by anubis2814** [comment]
Well shouldn't it just post one popup after all of them are uploaded?  I have a feeling one would have to post then edit to see which order they are in. Or the popup could show which order they are assigned in view thumbnails but that might be a real pain

**Comment by MrPetovan** [comment]
Everything is a pain. The way I add descriptions to multiple uploaded images is by first looking at a preview of the post to figure out which one is first.

**Comment by N-Upchurch** [comment]
> When uploading images to posts there's no option to add the caption/alt-text before posting outside of manually editing the BB-Code except when you go out of your way to upload first to the photo library and then explicitly reshare from there.

I have also run into this after joining a Friendica server, which led me here to see if there was an issue open about it . IMO this is a pretty urgent accessibility issue that most other federated social platforms have long since nailed. 

I know Friendica is a FLOSS project and work is done on a volunteer basis, but it is really disappointing to see that such a critical accessibility feature has gone unaddressed since 2023, especially from such an otherwise outstanding project.

and on this:

> Yeah it could come off as too judgy and self-righteous, trying to shame them

I reckon a user ought to feel some degree of shame when deciding to make their post inaccessible to folks 😆

**Comment by annando** [comment]
@N-Upchurch we do support alt texts. Only downside is that you have to enter it inside the BBCode. ALternatively you can upload the picture to the gallery and set a description there.

**Comment by N-Upchurch** [comment]
I understand, but my fear is that not being able to apply it with a GUI when posting introduces a great deal of friction to the addition of alt text and likely significantly reduces the use of alt text across Friendica, rendering the platform less accessible that its contemporaries. It's not an issue of function, it's an issue of UX.

**Comment by annando** [comment]
@N-Upchurch sure, that functionality should be improved. But sadly we don't have got any UI coders who could have got a look at it. There are some things that we do support from the backend without proper support on the frontend because of that problem.
